frontend/week4/js/homework2.js

```
@@ -0,0 +1,51 @@
+function errorMessage(errorTypeMe) {
+    document.getElementById("errorMessage").innerHTML += " Invalid " + errorTypeMe + "!! <br>";
+}
+
+
+function clearErrorMessage() {
+    document.getElementById("errorMessage").innerHTML = "";
+}
+
+function printAnswer(bmi, weightType) {
+    let name = document.getElementById("usersName").value;
+    document.getElementById("resultText").innerHTML = name + ", BMI = " + bmi + ", weight type = " + weightType;
+}
+
+function clearAnswer() {
+
+    document.getElementById("resultText").innerHTML = "";
+
+}
+
+function calculateWeightType(bmi) {
+    if (bmi < 20) {
+        return "under weight";
+    } else if (bmi > 20 && bmi < 30) {
+        return "normal weight";
+    } else {
+        return "over weight";
+    }
+
+}
+
+function calculateBmi() {
+    let length = document.getElementById("usersLength").value;
+    let weight = document.getElementById("usersweight").value;
+    if (length < 100 || length > 250) {
```

If there is an error the BMI is still calculated, I think the program should stop if the numbers are wrong.
You could `throw new Error("Length is invalid");` from this function, and catch it in the click-listener. Then `catch(e) { printError(e.message);}`